src/Data/Vec/Properties.agda

```diff
@@ -12,14 +12,16 @@ open import Algebra.Definitions
 open import Data.Bool.Base using (true; false)
 open import Data.Fin.Base as Fin using (Fin; zero; suc; toℕ; fromℕ<; _↑ˡ_; _↑ʳ_)
 open import Data.List.Base as List using (List)
+import Data.List.Properties as Listₚ
 open import Data.Nat.Base
 open import Data.Nat.Properties
-  using (+-assoc; m≤n⇒m≤1+n; ≤-refl; ≤-trans; suc-injective)
+  using (+-assoc; m≤n⇒m≤1+n; ≤-refl; ≤-trans; suc-injective; +-comm; +-suc)
 open import Data.Product.Base as Prod
   using (_×_; _,_; proj₁; proj₂; <_,_>; uncurry)
 open import Data.Sum.Base using ([_,_]′)
 open import Data.Sum.Properties using ([,]-map)
 open import Data.Vec.Base
+open import Data.Vec.Relation.Binary.Reasoning.Propositional as VecReasoning renaming (begin_ to begin′_; _∎ to _∎′)
```

Comment:
This is the one thing I can see straightaway which looks sub-optimal. You shouldn't need to rename the combinators you've worked hard to develop. There is an issue about overlapping reasoning system coexisting within a given development,so that is why [the style guide recommends](https://github.com/agda/agda-stdlib/blob/fd541d090bd2cb18f7eedd4c74d33fff2f953828/notes/style-guide.md?plain=1#L245-L254) that appeals to such things be localised to a proof, and that the tools themselves be packaged as a module for selective `open`ing.

Here, I would make the `open import` into an `import`, and then use `open VecReasoning` when you need the enhanced combinators... and that way, no renaming will be required!

Reply:
I moved the `open` statement to local `where` blocks. However, this library is intended to coexist with the standard propositional equational reasoning combinators, so `begin_` and QED still have conflicts as in <https://github.com/agda/agda-stdlib/pull/2067/commits/da40fb82556b0aa23cb61e0692223f475795b0df>.